### PR TITLE
Fix delegate retrieval when posting tx

### LINKED ISSUE #1734: Internal server error when storing a transaction
**Describe the bug**
An internal server error (500) is triggered sometimes when posting transactions (detected on Optimism chain)

Taking a look at the logs, it seems that is related to the delegates:

```python
2023-10-27 13:00:44,110 [ERROR] [MainProcess] Internal Server Error: /api/v1/safes/0x4ABa01FB8E1f6BFE80c56Deb367f19F35Df0f4aE/multisig-transactions/
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 56, in wrapper_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/app/safe_transaction_service/history/views.py", line 692, in post
    serializer.save()
  File "/app/safe_transaction_service/history/serializers.py", line 263, in save
    proposer = SafeContractDelegate.objects.get(
  File "/usr/local/lib/python3.10/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/query.py", line 640, in get
    raise self.model.MultipleObjectsReturned(
safe_transaction_service.history.models.SafeContractDelegate.MultipleObjectsReturned: get() returned more than one SafeContractDelegate -- it returned 3!
```

**Additional context**
- Related to https://github.com/5afe/safe-support/issues/315

### PR DESCRIPTION
**Pull request by Uxio0**
- Closes #1734

### CODE REVIEW

File: safe_transaction_service/history/models.py [hunk 1]
```diff
@@ -1715,11 +1715,23 @@ def created_block_number(self) -> Optional[Type[int]]:
...
-    def get_delegates_for_safe(self, address: ChecksumAddress) -> Set[ChecksumAddress]:
-        return set(
-            self.filter(safe_contract_id=address)
-            .values_list("delegate", flat=True)
-            .distinct()
+    def get_delegates_for_safe(
+        self, safe_address: ChecksumAddress
+    ) -> QuerySet["SafeContractDelegate"]:
+        return self.filter(
+            # If safe_contract is null on SafeContractDelegate, delegates are valid for every Safe
```
Comment (moisses89) [inline safe_transaction_service/history/models.py:1722]: Delegates are valid for every Safe which delegator is an owner. 
This method is returning every delegate where safe_contract is None without consider if the delegator is an owner of that Safe.

File: safe_transaction_service/history/models.py [hunk 2]
```diff
@@ -1751,7 +1760,7 @@ class SafeContractDelegate(models.Model):
         related_name="safe_contract_delegates",
         null=True,
         default=None,
-    )
+    )  # If not defined, delegate is valid for every Safe delegator is an owner of
```
Comment (moisses89) [inline safe_transaction_service/history/models.py:1763]:
```suggestion
    )  # If safe_contract is not defined, delegate is valid for every Safe which delegator is an owner
```

**Review by moisses89 — CHANGES_REQUESTED** [review]

File: safe_transaction_service/history/models.py [hunk 3]
```diff
@@ -1715,11 +1715,23 @@ def created_block_number(self) -> Optional[Type[int]]:
...
+    def get_delegates_for_safe(
+        self, safe_address: ChecksumAddress
+    ) -> QuerySet["SafeContractDelegate"]:
+        return self.filter(
+            # If safe_contract is null on SafeContractDelegate, delegates are valid for every Safe
+            Q(safe_contract_id=safe_address)
+            | Q(safe_contract=None)
+        )
+
+    def get_for_safe_and_delegate(
```
[the remark this answers is not in the document]
Comment (Uxio0) [reply safe_transaction_service/history/models.py:1727]: No, it's returning the class object, SafeContractDelegate. I feel `get` should be enough, if not it's very redundant and also not clear, as you are not getting delegates, you are getting the entire object

File: safe_transaction_service/history/models.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/history/models.py:1725]: Fixed

**Pull request merged**